sonar-plugin-api/src/main/java/org/sonar/api/batch/scm/ScmProvider.java

```diff
@@ -64,4 +64,19 @@ public BlameCommand blameCommand() {
   public Set<Path> branchChangedFiles(String targetBranchName, Path rootBaseDir) {
     return null;
   }
+
+  /**
+  * The relative path to the project from SCM root, typically ".".
+  */
+  public String projectBasedir() {
+    return ".";
+  }
+
+  /**
+   * The corresponding revision ID of the pull request,
+   * for example the SHA1 of the last commit in a Git branch.
+   */
+  public String revisionId() {
+    return "";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3400)